lib/authz/permissions.go

```diff
@@ -568,6 +568,7 @@ func roleSpecForProxy(clusterName string) types.RoleSpecV6 {
 				types.NewRule(types.KindSAMLIdPServiceProvider, services.RO()),
 				types.NewRule(types.KindUserGroup, services.RO()),
 				types.NewRule(types.KindIntegration, services.RO()),
+				types.NewRule(types.KindPlugin, services.RO()),
```

Comment:
Are they any security implications related to letting proxy reading plugins?

Reply:
`RO()` rule means Proxy can read credentials, which in most cases will be user-supplied (either static, or via OAuth short-lived tokens). It's true that there is no need to give such broad permissions in the general case (e.g. for other hosted plugins, Proxy does not need to read the credentials, only Auth).

I have modified this rule so we only give Proxy access to openai plugins. f84fa158d6